contracts/tezos/tezosSwap.py

```diff
@@ -0,0 +1,420 @@
+
+    """
+        Default entrypoint for 
+
+        args:
+            _active: boolean value [tru:active, false:inactive] representing contract state
+    """
+
+    @sp.entry_point
+    def default(self, unit):
+        pass
+       
+    """
+        Toggle contract active state
+
+        args:
+            _active: boolean value [tru:active, false:inactive] representing contract state
+    """ 
+    
+    @sp.entry_point
+    def toggleContractState(self, _active):
+        self.onlyByAdmin()
+        self.data.active = _active
+    
+    """
+        Update contract admin
+
+        args:
+            _admin: new admin addr
+    """
+
+    @sp.entry_point
+    def updateAdmin(self, _admin):
+        self.onlyByAdmin()
+        self.data.admin = _admin
+
+    """
+        Add or update swap pair
+
+        args:
+            _pair: A map of pairs
+    """
+
+    @sp.entry_point
+    def addSwapPair(self, _pairs):
+        self.onlyByAdmin()
+        sp.for pair in _pairs.keys():
+            self.data.pairs[pair] = _pairs[pair]
+
+    """
+        Remove swap pair
+
+        args:
+            _pairs: A list of pair names
+    """
+
+    @sp.entry_point
+    def removeSwapPair(self, _pairs):
+        self.onlyByAdmin()
+        sp.for pair in _pairs:
+            del self.data.pairs[pair]
+
+    """
+        Update swap fees
+
+        args:
+            _pairs: A list of pair names
+    """
+
+    @sp.entry_point
+    def updateSwapFees(self, _swapFee, _commission):
+        self.onlyByAdmin()
+        self.data.commission=_commission
+        self.data.swapFee=_swapFee
+
+    """
+        Sets baker for the contract
+
+        args:
+            _baker: A baker address
+    """
+
+    @sp.entry_point
+    def setBaker(self, _baker):
+        self.onlyByAdmin()
+        sp.set_delegate(_baker)
+
+    """
+        Transfers the accumulated service fee
+
+        args:
+            _to: Account address to transfer the fees
+    """
+
+    @sp.entry_point
+    def transferFees(self, _to):
+        self.onlyByAdmin()
+        sp.verify((sp.balance-sp.mutez(self.data.userXTZDeposit))>sp.mutez(0))
+        sp.send(_to, (sp.balance-sp.mutez(self.data.userXTZDeposit)))
+
+    """
+        Initiate new swap
+
+        args:
+            _swapHash: swap hash
+            _value: amount of tokens exchanged in the swap
+            _expected: amount of tokens expected in return
+            _refundTimestamp: unix time(sec) after which the swap expires
+            _pair: The swap pair being exchanged
+            _asset: Current token 
+    """
+
+    @sp.entry_point
+    def offer(self, _swapHash, _value, _expected, _refundTimestamp, _pair, _asset):
+        self.isInitiable(_swapHash, _refundTimestamp, _pair, _asset)
+        sp.if _asset=="xtz":
+            sp.verify(sp.amount==sp.mutez(_value+self.data.commission),"AMOUNT_MISMATCH")
+            self.data.userXTZDeposit+= _value+self.data.commission
```

Comment:
perhaps just use `sp.amount` here

Reply:
sp.amount needs to be converted to nat we don't have to do that conversion as _value and sp.amount will be the same as we already have a check for that.